packages/angular-material/test/object-control.spec.ts

```diff
@@ -0,0 +1,209 @@
+  THE SOFTWARE.
+*/
+import { NgRedux, NgReduxModule } from '@angular-redux/store';
+import { MockNgRedux } from '@angular-redux/store/lib/testing';
+import { CommonModule } from '@angular/common';
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { ReactiveFormsModule } from '@angular/forms';
+import { MatCardModule, MatFormFieldModule, MatInputModule } from '@angular/material';
+import { BrowserDynamicTestingModule } from '@angular/platform-browser-dynamic/testing';
+import { NoopAnimationsModule } from '@angular/platform-browser/animations';
+import { JsonFormsModule } from '@jsonforms/angular';
+import { ControlElement } from '@jsonforms/core';
+import {
+  TextControlRenderer,
+  TextControlRendererTester,
+  VerticalLayoutRenderer,
+  verticalLayoutTester
+} from '../src';
+import { ObjectControlRenderer, ObjectControlRendererTester } from '../src/other/object.renderer';
+
+const uischema1: ControlElement = { type: 'Control', scope: '#' };
+const uischema2: ControlElement = { type: 'Control', scope: '#/properties/foo' };
+const schema1 = {
+  type: 'object',
+  properties: {
+    foo: { type: 'string' }
+  }
+};
+const schema2 = {
+  type: 'object',
+  properties: {
+    foo: {
+      type: 'object',
+      properties: {
+        foo_1: { type: 'string' }
+      }
+    },
+    bar: {
+      type: 'object',
+      properties: {
+        bar_1: { type: 'string' }
+      }
+    }
+  }
+};
+const renderers = [
+  { tester: TextControlRendererTester, renderer: TextControlRenderer },
+  { tester: verticalLayoutTester, renderer: VerticalLayoutRenderer },
+];
+
+describe('Object Control tester', () => {
+  it('should succeed', () => {
+    expect(ObjectControlRendererTester(uischema1, schema1)).toBe(2);
+    expect(ObjectControlRendererTester(uischema2, schema2)).toBe(2);
+  });
+});
+describe('Object Control', () => {
+  let fixture: ComponentFixture<any>;
+  let component: any;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        ObjectControlRenderer,
+        TextControlRenderer,
+        VerticalLayoutRenderer
+      ],
+      imports: [
+        CommonModule,
+        JsonFormsModule,
+        NgReduxModule,
+        MatCardModule,
+        NoopAnimationsModule,
+        MatFormFieldModule,
+        MatInputModule,
+        ReactiveFormsModule
+      ],
+      providers: [
+        { provide: NgRedux, useFactory: MockNgRedux.getInstance }
+      ]
+    }).overrideModule(BrowserDynamicTestingModule, {
+      set: {
+          entryComponents: [
+            TextControlRenderer,
+            VerticalLayoutRenderer
+          ]
+      }
+  })
+  .compileComponents();
+
+    MockNgRedux.reset();
+    fixture = TestBed.createComponent(ObjectControlRenderer);
+    component = fixture.componentInstance;
+  }));
+
+  it('render all elements', async(() => {
+    const mockSubStore = MockNgRedux.getSelectorStub();
```

Comment:
If the `setupMockStore` method would receive the next state instead of the triplet of schema/UI schema/data, we could re-use it here to avoid some boilerplate

Reply:
We should double check the tests anyway. I'll open an issue to recheck this.